google-cloud-pubsub/LOGGING.md

```diff
@@ -30,3 +33,87 @@ module GRPC
   extend MyLogger
 end
 ```
+
+## Adding gRPC interceptors
+
+[gRPC](https://github.com/grpc/grpc/tree/master/src/ruby) accepts [Ruby-language
+interceptors](https://github.com/grpc/proposal/blob/master/L11-ruby-interceptors.md) that allow you to insert your own
+custom logging into a client's RPC calls. (gRPC interceptors are also useful for auth, metrics, tracing and similar
+use cases.)
+
+This library performs RPCs using the following [gapic](https://github.com/googleapis/gapic-generator-ruby) clients from
+the underlying
+[google-cloud-pubsub-v1](https://github.com/googleapis/google-cloud-ruby/tree/main/google-cloud-pubsub-v1) library:
+
+* [`Google::Cloud::PubSub::V1::IAMPolicy::Client`](https://googleapis.dev/ruby/google-cloud-pubsub-v1/latest/Google/Cloud/PubSub/V1/IAMPolicy/Client.html)
+* [`Google::Cloud::PubSub::V1::Publisher::Client`](https://googleapis.dev/ruby/google-cloud-pubsub-v1/latest/Google/Cloud/PubSub/V1/Publisher/Client.html)
+* [`Google::Cloud::PubSub::V1::SchemaService::Client`](https://googleapis.dev/ruby/google-cloud-pubsub-v1/latest/Google/Cloud/PubSub/V1/SchemaService/Client.html)
+* [`Google::Cloud::PubSub::V1::Subscriber::Client`](https://googleapis.dev/ruby/google-cloud-pubsub-v1/latest/Google/Cloud/PubSub/V1/Subscriber/Client.html)
+
+To add a gRPC interceptor to one or more of these clients, first implement your logic as a subclass of
+[`GRPC::ClientInterceptor`](https://www.rubydoc.info/gems/grpc/GRPC/ClientInterceptor).
```

Comment:
Can you add a sentence describing what functionality the interceptor below implements? (e.g. "The example below logs all gRPC calls and sets a metadata field"). So it's a bit more clear what's required by the interceptor contract and what's there just because we decided to do it.

Reply:
Added, thanks!